packages/engine/src/avatar/functions/avatarFunctions.ts

```diff
@@ -224,8 +226,12 @@ export const retargetAvatarAnimations = (entity: Entity) => {
  */
 export const setAvatarSpeedFromRootMotion = () => {
   const manager = getState(AnimationState)
-  const run = manager.loadedAnimations[preloadedAnimations.locomotion].animations[4] ?? [new AnimationClip()]
-  const walk = manager.loadedAnimations[preloadedAnimations.locomotion].animations[6] ?? [new AnimationClip()]
+  const animations = getComponent(
+    manager.loadedAnimations[preloadedAnimations.locomotion],
+    AnimationComponent
+  ).animations
+  const run = AnimationClip.findByName(animations, 'Run_RootMotion')
```

Comment:
let's make a note here to hande this generically in the future, and also put this on a module level const

Reply:
yeah this will need to be redone with upcoming animation system refactors, avatar animations should generally be handled generically, right now everything is magic strings and magic numbers.